libraries/AP_Motors/AP_Motors_Class.cpp

```diff
@@ -199,7 +199,17 @@ void AP_Motors::add_motor_num(int8_t motor_num)
     if (motor_num >= 0 && motor_num < AP_MOTORS_MAX_NUM_MOTORS) {
         uint8_t chan;
         SRV_Channel::Aux_servo_function_t function = SRV_Channels::get_motor_function(motor_num);
-        SRV_Channels::set_aux_channel_default(function, motor_num);
+#if HAL_WITH_IO_MCU
+        // can't do digital outputs with IOMCU, so remap to AUX if possible
+        if (is_digital_pwm_type() && motor_num < 6) {
```

Comment:
If I have a copter with motors than 6 motors I think this will do odd stuff?
motor number, channel
0  -> 8
1   -> 9
2   -> 10
3   -> 11
4   -> 12
5   ->13
6   ->6
7   -> 7
8   -> 8
9   -> 9
10 -> 10
11 -> 11

So we will end no defaults set for motors 0 to 3 because there overwritten by 8 to 11?

Reply:
Don't think they can be overwritten because they were assigned earlier

Reply:
Right, so you would just not get  8 to 11. Which I guess is not so bad. 

I think we we added a arming check to make sure you cant arm unless you have the correct motors assigned that might be OK.